app/models/wordfilter.rb

```diff
@@ -0,0 +1,22 @@
+class Wordfilter < ApplicationRecord
+  flag :locations, %i[post comment reaction]
+  enum action: {
+    censor: 10,
+    report: 20,
+    hide: 30,
+    reject: 40
+  }, _prefix: 'action_'
+
+  validates :pattern, presence: true
+
+  scope :matching, ->(text) {
+    # Match regexes with ~* and create an exact match via ILIKE
+    where("? ~* ('.*' || pattern || '.*')", text).where(regex_enabled: true).or(
+      where("? ILIKE ('%' || pattern || '%')", text).where(regex_enabled: false)
+    )
+  }
+
+  def self.action_for(location, text)
+    matching(text).where_locations(location).order(action: :desc).first&.action&.to_sym
```

Comment:
why are you using `where_locations` instead of `where(locations: location)`?

Reply:
Because active flag doesn't override where() it instead exposes this helper method